From 9161e5c7782bb94dfcfe2e8298620b2b378bfabf Mon Sep 17 00:00:00 2001
From: FrancescoFerraro <francesco.f@variscite.com>
Date: Fri, 11 Nov 2022 10:51:08 +0100
Subject: [PATCH 10/12] imx8mm_var_dart: adjust ddrc_opp_table for DDR4 on
 VAR-SOM-MX8M-MINI

The frequencies in the ddrc_opp_table must be aligned with the fsp rate
values that U-Boot use doing DDR4 calibration.

The fsp_table for VAR-SOM-MX8M-MINI is https://github.com/varigit/uboot-imx/blob/f931cdebf9830c1ded3c0c4ac58d564c695e551c/board/variscite/imx8mm_var_dart/ddr4_timing.c#L539
so we have two available frequencies calculated as:
opp-266M = 1066M/4
opp-600M = 2400M/4

Signed-off-by: FrancescoFerraro <francesco.f@variscite.com>
(cherry picked from commit aa29c791f95c7f3a71ffa339053e92284abc50e1)
Signed-off-by: Vanessa Maegima <vanessa.maegima@foundries.io>
---
 arch/arm/dts/imx8mm-var-som.dtsi | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/arch/arm/dts/imx8mm-var-som.dtsi b/arch/arm/dts/imx8mm-var-som.dtsi
index f0bf27ac3e8..f8e6395a3a1 100644
--- a/arch/arm/dts/imx8mm-var-som.dtsi
+++ b/arch/arm/dts/imx8mm-var-som.dtsi
@@ -42,16 +42,12 @@
 	ddrc_opp_table: opp-table {
 		compatible = "operating-points-v2";
 
-		opp-25M {
-			opp-hz = /bits/ 64 <25000000>;
+		opp-266M {
+			opp-hz = /bits/ 64 <266500000>;
 		};
 
-		opp-100M {
-			opp-hz = /bits/ 64 <100000000>;
-		};
-
-		opp-750M {
-			opp-hz = /bits/ 64 <750000000>;
+		opp-600M {
+			opp-hz = /bits/ 64 <600000000>;
 		};
 	};
 };
-- 
2.34.1

